src/main/java/Helper/DateHandler.java

```diff
@@ -0,0 +1,195 @@
+
+    // stores the length of date argument.
+    private static final int DATE_LENGTH = 10;
+
+    // stores the length of time argument.
+    private static final int TIME_LENGTH = 4;
+
+    // stores the time argument.
+    private final LocalTime time;
+
+    /**
+     * Constructor for DateHandler class.
+     *
+     * @param date the string representation of the date.
+     * returns an instance of DateHandler.
+     */
+    public DateHandler(String date) {
+        date = date.trim();
+        if (date.contains(SPACE)) {
+            this.date = LocalDate.parse(date.split(SPACE, LIMIT)[DATE_IDX]);
+            String hour = date.split(SPACE, LIMIT)[TIME_IDX].substring(START_INDEX, MINUTE_INDEX);
+            String minute = date.split(SPACE, LIMIT)[TIME_IDX].substring(MINUTE_INDEX);
+            this.time = LocalTime.of(Integer.valueOf(hour), Integer.valueOf(minute));
+            this.isTimeGiven = true;
+        } else {
+            this.date = LocalDate.parse(date);
+            this.time = LocalTime.MIN;
+            this.isTimeGiven = false;
+        }
+    }
+
+    /**
+     * Constructor for DateHandler.
+     *
+     * @param date the date associated with the task from the previous runs.
+     * @param dummyVariable to differentiate between the constructors.
+     */
+    public DateHandler(String date, int dummyVariable) {
+
+        String[] strings = date.split(SPACE);
+
+        for (int i = 0; i < strings.length / 2; i++) {
+            String temp = strings[i];
+            strings[i] = strings[strings.length - i - 1];
+            strings[strings.length - i - 1] = temp;
+        }
+
+        int year_idx = 0;
+        int month_idx = 1;
+        int day_idx = 2;
+        this.date = LocalDate.of(Integer.valueOf(strings[year_idx]), Month.valueOf(strings[month_idx]),
+                Integer.valueOf(strings[day_idx]));
+
+        int splitArrLenWithTime = 5;
+        if (strings.length == splitArrLenWithTime) {
+            //extract time
+            String[] time = strings[strings.length - 2].split(":");
+            int hour_idx = 0;
+            int minute_idx = 1;
+            int hour = Integer.valueOf(time[hour_idx]);
+            int minute = Integer.valueOf(time[minute_idx].substring(1));
+            this.time = LocalTime.of(hour, minute);
+            this.isTimeGiven = true;
+        } else {
+            this.time = LocalTime.MIN;
+            this.isTimeGiven = false;
+        }
+    }
+
+    /**
+     * checks if the userDate is of the valid format.
+     *
+     * @param userDate the date input by user.
+     * @throw WrongDateArgumentException if the user inputs an invalid date.
+     * @throw WrongTimeArgumentException if the user inputs an invalid time.
+     */
+    public static void checkValidDate(String userDate) {
+
+        // handle time checking
+        if (userDate.contains(SPACE)) {
+            String hr_mt = userDate.split(SPACE, LIMIT)[TIME_IDX].trim();
```

Comment:
Perhaps the String `hr_mt` could be renamed to something clearer. This could be to ensure there is no ambiguity.